core/src/test/java/io/hyperfoil/core/client/HttpDecodeSpaceTest.java

```diff
@@ -0,0 +1,206 @@
+                           if (status != 200) {
+                              ctx.fail("Status Code : " + status);
+                           } else {
+                              statusReceived.set(true);
+                           }
+                        })
+                        .onCompletion(s -> {
+                           if (statusReceived.get()) {
+                              async.complete();
+                           } else {
+                              ctx.fail("Status was not received.");
+                           }
+                        }).build();
+                  request.method = HttpMethod.GET;
+                  request.path = path;
+                  request.start(handlers, new SequenceInstance(), new Statistics(System.currentTimeMillis()));
+
+                  client.next().request(request, null, true, null, false);
+               });
+            } catch (Exception e) {
+               ctx.fail(e);
+            }
+         }
+      });
+   }
+
+   private HttpClientPool client(int port, boolean ssl, HttpVersion[] versions) throws Exception {
+      HttpBuilder builder = HttpBuilder.forTesting()
+            .protocol(ssl ? Protocol.HTTPS : Protocol.HTTP).host("localhost").port(port);
+      builder.allowHttp2(Stream.of(versions).anyMatch(v -> v == HttpVersion.HTTP_2_0));
+      builder.allowHttp1x(Stream.of(versions).anyMatch(v -> v == HttpVersion.HTTP_1_1));
+      return HttpClientPoolImpl.forTesting(builder.build(true), 1);
+   }
+
+   private void server(boolean ssl, List<io.vertx.core.http.HttpVersion> serverVersions, Handler<AsyncResult<HttpServer>> handler) {
+      HttpServer httpServer;
+      if (ssl) {
+         JksOptions keyStoreOptions = new JksOptions().setPath("keystore.jks").setPassword("test123");
+         HttpServerOptions httpServerOptions = new HttpServerOptions()
+               .setSsl(true)
+               .setKeyStoreOptions(keyStoreOptions)
+               .setUseAlpn(true)
+               .setAlpnVersions(serverVersions);
+         httpServer = vertx.createHttpServer(httpServerOptions);
+         httpServer.requestHandler(HttpDecodeSpaceTest::isPathCorrect).listen(0, "localhost", handler);
+      } else {
+         httpServer = vertx.createHttpServer();
+         httpServer.requestHandler(HttpDecodeSpaceTest::isPathCorrect).listen(0, "localhost", handler);
+      }
+   }
+
+   private static void isPathCorrect(HttpServerRequest req) {
+      int status = 0;
+      String url = req.uri();
+      if (url.contains(" ")) {
+          status = 600;
+      } else {
+          int question = url.indexOf("?");
+          String subFirst = "";
+          String subSecond = "";
+          if (question != -1) {
+              subFirst = url.substring(0 , question);
+              subSecond = url.substring(url.lastIndexOf("?") + 1);
+              if (subFirst.contains("+")) {
+                  status = 601;
+              } else if (subFirst.contains("%20")) {
```

Comment:
This condition is superfluous (status becomes 200 anyway).

Reply:
I have removed unnecessary if conditions.  Please review.